src/marqo/config.py

```diff
@@ -43,5 +46,15 @@ def set_url(self, url):
             self.cluster_is_remote = True
             if "s2search.io" in lowered_url:
                 self.cluster_is_s2search = True
+            if "api.marqo.ai" in lowered_url:
+                self.cluster_is_marqo = True
+                self.marqo_url_resolver = MarqoUrlResolver(api_key=self.api_key, expiration_time=60)
         self.url = url
         return self.url
+
+    def get_url(self, index_name=None,):
+        if not self.cluster_is_marqo:
+            return self.url
+        if self.cluster_is_marqo and not index_name:
+            return self.url + "/api"
+        return self.marqo_url_resolver.urls_mapping[index_name]
```

Comment:
do we need error handling if `index_name` not in `urls_mapping`

Reply:
Mindset was to keep client dumb and let api handle it, but in that case we indeed need to handle it with client, wip on that one

Reply:
Is it reasonable to raise Exception(f"Index {item} is still being created") when index is creating
and Exception(f"Index {item} does not exist") or should it be softer and not stopping the code?

Reply:
Yes I agree that we need to keep the client as simple as possible, but we may have no choice here. 

In terms of error message we should be as clear as possible.

For example something like this (feel free to modify this if needed) :

"The Python client could not resolve the endpoint for the index name `my-index`. This could be due to the index is still in the process of being created, or that the client's cache has not yet been updated. 
- please try again in a couple of minutes, to allow the client's cache to be updated
- you can also try the querying the index's status: GET https://<INSERT INDEX ENDPOINT HERE>/status
- If the issue still persists please make a Marqo cloud support ticket here <LINK

Reply:
Yes I agree that we need to keep the client as simple as possible, but we may have no choice here. 

In terms of error message we should be as clear as possible.

For example something like this (feel free to modify this if needed) :

"The Python client could not resolve the endpoint for the index name `my-index`. This could be due to the index is still in the process of being created, or that the client's cache has not yet been updated. 
- please try again in a couple of minutes, to allow the client's cache to be updated
- you can also try the querying the index's status: GET https://<INSERT INDEX ENDPOINT HERE>/status
- If the issue still persists please make a Marqo cloud support ticket here <LINK